### PR TITLE
Add NFC_NXP_CHIP_TYPE keyword for detecting NXP NFC chips on newer phone models

### PR DESCRIPTION
**Pull request by weebl2000**
Add new NXP chip key value for SN1x0 and SN2x0 NXP chip types

### CODE REVIEW

**Comment by roussosalex** [comment]
Thanks for contributing! We already added the new chip codes in the `v2-dev` branch. If you would like to re-target this PR to the `v2-dev` branch for the new nfc chip keyword, please see the changes requested below.

File: app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java [hunk 1]
```diff
@@ -66,8 +68,7 @@ protected boolean onLine(String line, NfcChipGuess guess) {
                 guess.improveConfidence(0.9f);
                 if (guess.chipName == null)
                     guess.chipName = "NXP Device " + formatNXPDeviceNode(keyVal.second);
-            }
-            else if ("NXP_NFC_CHIP".equals(keyVal.first)) {
+            } else if ("NXP_NFC_CHIP".equals(keyVal.first) || "NXP_NFC_CHIP_TYPE".equals(keyVal.first)) {
```
Comment (roussosalex) [inline app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java]: Please make the list of keywords a  "Set<String>" alongside the NXPChipMap for future extendability, thanks!

**Comment by weebl2000** [comment]
> Thanks for contributing! We already added the new chip codes in the `v2-dev` branch. If you would like to re-target this PR to the `v2-dev` branch for the new nfc chip keyword, please see the changes requested below.

Done

**Review by roussosalex — CHANGES_REQUESTED** [review]
For some reason this PR is shown as having conflicts with the `v2-dev` branch.

File: app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java (hunk 1 above)
Comment (weebl2000) [reply app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java]: Done!

**Comment by weebl2000** [comment]
> For some reason this PR is shown as having conflicts with the `v2-dev` branch.

My original commit had SN1x0, SN2x0 instead of SN1X0, SN2X0 as strings, I did fix the conflict though, weird that it still showed conflicts. Rebased on v2-dev with the last change - should be good now for sure.

File: app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java [hunk 2]
```diff
@@ -36,6 +39,15 @@ public class NXPDetector extends BaseConfigLineDetector {
         put("0x0C","SN2X0");
     }};
 
+    // Different versions of libnfc use different keywords
+    private final static Set<String> KNOWN_NXP_NFC_CHIP_KEYWORDS;
```
Comment (roussosalex) [inline app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java]: Please adapt your code to the surrounding code-style. Use a shorter name, use UpperCamelCase (all our constants should start with uppercase letters, all variables use camelCase, therefore UpperCamelCase for this one) as in `NXPChipMap` and then use static-initialization (the same type used in `NXPChipMap` and explained [here](https://stackoverflow.com/a/2041789))

**Review by roussosalex — CHANGES_REQUESTED** [review]

**Comment by roussosalex** [comment]
> should be good now for sure

Looks good, no more conflicts. I had to request one more change unfortunately, sorry!

File: app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java (hunk 2 above)
Comment (weebl2000) [reply app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java]: > Please adapt your code to the surrounding code-style. Use a shorter name, use UpperCamelCase (all our constants should start with uppercase letters, all variables use camelCase, therefore UpperCamelCase for this one) as in `NXPChipMap` and then use static-initialization (the same type used in `NXPChipMap` and explained [here](https://stackoverflow.com/a/2041789))

I can do this, but the static naming convention for Java is all caps and underscores. https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

image

I would have used `Set.of()` notation usually, but it's not suited for SDK version 19.

Comment (roussosalex) [reply app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java]: > I can do this

Thanks

Comment (roussosalex) [reply app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java]: > Use a shorter name

One small step for this commit, one giant leap for this PR.

Comment (weebl2000) [reply app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/NXPDetector.java]: > > Use a shorter name
> 
> One small step for this commit, one giant leap for this PR.

Done, do I have to pick up the can too? 😄

**Review by roussosalex — APPROVED** [review]
Great PR, fits incredibly well into the existing code-style!

**Comment by aswin-devil** [comment]
8

**Comment by roussosalex** [comment]
> 8

?

**Pull request merged**